frontend/src/features/admin-form/create/workflow/components/WorkflowContent/EditStepBlock/ApprovalsBlock.tsx

```diff
@@ -0,0 +1,140 @@
+import { useCallback, useState } from 'react'
+import { Controller, UseFormReturn } from 'react-hook-form'
+import { FormControl } from '@chakra-ui/react'
+
+import { SingleSelect } from '~components/Dropdown'
+import FormErrorMessage from '~components/FormControl/FormErrorMessage'
+import FormLabel from '~components/FormControl/FormLabel'
+import Toggle from '~components/Toggle'
+
+import { BASICFIELD_TO_DRAWER_META } from '~features/admin-form/create/constants'
+
+import { useAdminFormWorkflow } from '../../../hooks/useAdminFormWorkflow'
+import { EditStepInputs } from '../../../types'
+
+import { EditStepBlockContainer } from './EditStepBlockContainer'
+
+interface ApprovalsBlockProps {
+  formMethods: UseFormReturn<EditStepInputs>
+  stepNumber: number
+}
+
+const APPROVAL_FIELD_NAME = 'approval_field'
+const FIELDS_TO_EDIT_NAME = 'edit'
+export const ApprovalsBlock = ({
+  formMethods,
+  stepNumber,
+}: ApprovalsBlockProps): JSX.Element => {
+  const {
+    control,
+    setValue,
+    formState: { errors },
+    clearErrors,
+    watch,
+  } = formMethods
+  const selectedApprovalField = watch(APPROVAL_FIELD_NAME)
+  const selectedEditFields = watch(FIELDS_TO_EDIT_NAME)
```

Comment:
Didn't see where is the corresponding form field that this name is tagged to.

Reply:
yes, this is being used: 
- selectedApprovalField is used to show and enable/disable the toggle upon load from db (ie, when the admin clicks on the edit step and visits the activeStepBlock)  
- selectedEditFields is used for validation of the react-hook-form state before allowing submit